### PR TITLE
Extend view and cell templates

### PR DESCRIPTION
**Pull request by mikolasstuchlik**
This pull request expands views and cell templates.

Motivation: Templates lack commonly used patterns like stand alone view or collection view elements. 

Updates:
* Add view template for Nibable protocol under FuntastyKit
* Add CollectionView Cell and Reusable views under CellKit/UIKit
* Rename Cell to TableCell
* Set default iPhone as iPhone X (iphone5_9) across modified templates
* Add template origin notice to source file headers across modified templates
* Add safe area margins by default across modified templates

[Regression warning]: I've made a mistake during development. I've started my work on outdated versions. I've already removed a lot of unwanted regression, but I ask reviewers to kindly seek for any remaining regression.

### CODE REVIEW

File: CollectionCell.xctemplate/TemplateInfo.plist [hunk 1]
```diff
@@ -0,0 +1,87 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>DefaultCompletionName</key>
+	<string>Collection Cell</string>
+	<key>Summary</key>
+	<string>This generates a new collection view cell view under CellKit.</string>
```
Comment (mkj-is) [inline CollectionCell.xctemplate/TemplateInfo.plist]: I think the second `view` is redundant there.

```suggestion
	<string>This generates a new collection view cell under CellKit.</string>
```

File: CollectionCell.xctemplate/TemplateInfo.plist [hunk 2]
```diff
@@ -0,0 +1,87 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>DefaultCompletionName</key>
+	<string>Collection Cell</string>
```
Comment (mkj-is) [inline CollectionCell.xctemplate/TemplateInfo.plist]: I prefer something like `CellKit Collection View Cell.xctemplate` so it is clear and the naming is consistent with other templates (even if it's longer). Also the names in comments should be the same as the name of the template.

```suggestion
	<string>CellKit Collection View Cell</string>
```

Sadly, this name is not what is showing in the list of templates in Xcode.

File: CollectionCell.xctemplate/___VARIABLE_sourceCellIdentifier___CollectionCell.swift [hunk 3]
```diff
@@ -0,0 +1,18 @@
+//
+//  ___FILENAME___
+//  ___PROJECTNAME___
+//
+//  Created by ___FULLUSERNAME___ on ___DATE___ using Collection View Cellkit Template (v1.0).
```
Comment (mkj-is) [inline CollectionCell.xctemplate/___VARIABLE_sourceCellIdentifier___CollectionCell.swift]: Also, CellKit has large K. 🙂

```suggestion
//  Created by ___FULLUSERNAME___ on ___DATE___ using CellKit Collection View Cell Template (v1.0).
```

File: CollectionCell.xctemplate/___VARIABLE_sourceCellIdentifier___CollectionCellModel.swift [hunk 4]
```diff
@@ -0,0 +1,17 @@
+//
+//  ___FILENAME___
+//  ___PROJECTNAME___
+//
+//  Created by ___FULLUSERNAME___ on ___DATE___ using Collection View Cellkit Template (v1.0).
```
Comment (mkj-is) [inline CollectionCell.xctemplate/___VARIABLE_sourceCellIdentifier___CollectionCellModel.swift]:
```suggestion
//  Created by ___FULLUSERNAME___ on ___DATE___ using CellKit Collection View Cell Template (v1.0).
```

File: CollectionReusableView.xctemplate/___VARIABLE_sourceViewIdentifier___CollectionReusableView.swift [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+//  ___FILENAME___
+//  ___PROJECTNAME___
+//
+//  Created by ___FULLUSERNAME___ on ___DATE___ using Collection Reusable View Template (v1.0).
+//  Copyright © ___YEAR___ ___ORGANIZATIONNAME___. All rights reserved.
+//
+
+import UIKit
+
+final class ___VARIABLE_sourceViewIdentifier___CollectionReusableView: UICollectionReusableView {
```
Comment (mkj-is) [inline CollectionReusableView.xctemplate/___VARIABLE_sourceViewIdentifier___CollectionReusableView.swift]: Similar file can be created using the `Cocoa Touch Class` template. Basically, the only difference is the `final` keyword. I think this don't belong to our templates.

File: TableCell.xctemplate/TemplateInfo.plist [hunk 6]
```diff
@@ -3,111 +3,82 @@
 <plist version="1.0">
 <dict>
 	<key>DefaultCompletionName</key>
-	<string>Cell</string>
-	
+	<string>Table Cell</string>
 	<key>Summary</key>
-	<string>This generates a new scene using The Funtasty MVVM-C architecture.</string>
-
+	<string>This generates a new table cell view for CellKit.</string>
```
Comment (mkj-is) [inline TableCell.xctemplate/TemplateInfo.plist]: This order of words is more traditional due to the name of the superclass.

```suggestion
	<string>This generates a new table view cell for CellKit.</string>
```

**Review by mkj-is — CHANGES_REQUESTED** [review]

File: CollectionCell.xctemplate/TemplateInfo.plist (hunk 2 above)
Comment (mikolasstuchlik) [reply CollectionCell.xctemplate/TemplateInfo.plist]: I see your point but I think, that this name is way too long. Also, I think, that the CellKit is redundant and inconsistent. I would object, why our other templates does not contain similar prefixes (for example: FuntastyKitCoordinator, FuntastyKitViewModel etc.). I'm fine with the word "View".

File: CollectionReusableView.xctemplate/___VARIABLE_sourceViewIdentifier___CollectionReusableView.swift (hunk 5 above)
Comment (mikolasstuchlik) [reply CollectionReusableView.xctemplate/___VARIABLE_sourceViewIdentifier___CollectionReusableView.swift]: While developing Angelcam, I found out, that I spend time by creating files and naming them consistently. This template reduces amount of time people spend by searching for typos in multiple gaps across multiple files. However, if you don't consider this template as something worth merging, I'm fine with forking this repository and having this template separate. :)

File: CollectionCell.xctemplate/TemplateInfo.plist (hunk 2 above)
Comment (mkj-is) [reply CollectionCell.xctemplate/TemplateInfo.plist]: Next step is dividing these templates into the frameworks, so we can lose the framework prefixes then.

Comment (mikolasstuchlik) [reply CollectionCell.xctemplate/TemplateInfo.plist]: :( You're the boss :D

Comment (mikolasstuchlik) [reply CollectionCell.xctemplate/TemplateInfo.plist]: But still, there will be some files created by this template by the time we divide this repo and those files won't be affected, so there will be some files with 5-7 words long names, which I consider a bit unfortunate.

File: CollectionReusableView.xctemplate/___VARIABLE_sourceViewIdentifier___CollectionReusableView.swift (hunk 5 above)
Comment (mikolasstuchlik) [reply CollectionReusableView.xctemplate/___VARIABLE_sourceViewIdentifier___CollectionReusableView.swift]: The Nibable protocol requires specific naming of multiple files and classes. There I see the difference between stock template and this one.

File: CollectionCell.xctemplate/TemplateInfo.plist (hunk 2 above)
Comment (mikolasstuchlik) [reply CollectionCell.xctemplate/TemplateInfo.plist]: I totally missed your point. I thought you want the CellKit string in name of files. Silly me.

File: CellKit/TableViewHeaderFooter.xctemplate/___VARIABLE_sourceCellIdentifier___TableViewHeaderFooter.swift [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+//
+
+import CellKit
+
+final class ___VARIABLE_sourceCellIdentifier___TableViewHeaderFooter: UITableViewHeaderFooterView {
+
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        
+        // Arbitary configuration here
```
Comment (mkj-is) [inline CellKit/TableViewHeaderFooter.xctemplate/___VARIABLE_sourceCellIdentifier___TableViewHeaderFooter.swift]: Tento komentář bych dal pryč, myslím si, že je to dost jasné i bez něj.

File: MVVM-C/NibableView.xctemplate/___VARIABLE_sourceViewIdentifier___View.swift [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+import FuntastyKit
+import UIKit
+
+final class ___VARIABLE_sourceViewIdentifier___View: UIView, Nibable {
+
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        
+        // Arbitrary configuration here.
```
Comment (mkj-is) [inline MVVM-C/NibableView.xctemplate/___VARIABLE_sourceViewIdentifier___View.swift]: Tento komentář bych taky dal pryč.

File: Makefile [hunk 9]
```diff
@@ -1,9 +1,18 @@
-TEMPLATE_DIR = $(HOME)/Library/Developer/Xcode/Templates/MVVM-C\ Templates
 
-install: $(TEMPLATE_DIR)
-	cp -R *.xctemplate $(TEMPLATE_DIR)
+TEMPLATE_NAMES = 'MVVM-C' 'CellKit'
```
Comment (mkj-is) [inline Makefile:2]: Nešlo by to nějak zjednodušit a vzít všechny `*/*.xctemplate` soubory a zkopírovat je?

**Review by mkj-is — APPROVED** [review]

**Pull request merged**